src/toniarts/openkeeper/world/MapLoader.java

```diff
@@ -85,18 +88,18 @@ public Spatial load(AssetManager assetManager, KwdFile object) {
 
         // Go through the map
         int tilesCount = object.getWidth() * object.getHeight();
-        Map[][] tiles = object.getTiles();
-        for (int x = 0; x < object.getWidth(); x++) {
-            for (int y = 0; y < object.getHeight(); y++) {
+        Tile[][] tiles = object.getTiles();
+        for (int width = 0; width < object.getWidth(); width++) {
+            for (int height = 0; height < object.getHeight(); height++) {

```

Comment:
And while at it, I don't understand the need to change the variable names here. As this really IS a coordinate in 2D system, and x & y are totally valid in that sense. "height" and "width" really aren't as much valid. But I'll let you decide.